lib/ansible/modules/system/pkt_steering.py

```diff
@@ -0,0 +1,388 @@
+                if intern_cpus is None:
+                    return intern_cpus
+                mask = self._form_per_node_bitmask(intern_cpus)
+                intern_qu_mask[qu] = mask
+
+        return intern_qu_mask
+
+    def form_qu_to_mask(self, arch_specs, numa_to_qus):
+        qu_to_mask = {}
+        for n, qus in numa_to_qus.items():
+            for qu in range(qus[0], qus[1]):
+                mask = self._form_per_node_bitmask(arch_specs[n])
+                qu_to_mask[qu] = mask
+
+        return qu_to_mask
+
+    def _expand_ext_cpus(self, cpus):
+        int_cpus = []
+
+        for repres in cpus:
+            if isinstance(repres, int):
+                int_cpus.append(repres)
+            elif isinstance(repres, list):
+                if len(repres) > 2:
+                    return None
+                try:
+                    int_cpus.extend(range(repres[0], repres[1] + 1))
+                except:
+                    return None
+            else:
+                return None
+
+        return int_cpus
+
+    def _form_per_node_bitmask(self, cpus):
+        bitmask = 0
+        for cpu in cpus:
+            bitmask = bitmask | (1 << int(cpu))
+
+        out = "%x" % bitmask
+        out_l = len(out)
+        if out_l > 8:
+            left_marg = out_l % 8
+            sub_grps = (out_l / 8) - 1
+            idx_l = [left_marg]
+            idx_l.extend(range(left_marg, left_marg + 8 * sub_grps, 8))
+            for idx in idx_l:
+                out = out[:idx] + "," + out[idx:]
+
+        return out
+
+
+def run(module, args):
+
+    interactor = KernelInteractor(module)
+
+    changed = False
+
+    if args['tech'] == 'rfs':
+        if args['state'] == 'absent':
+            changed = interactor.out_cnfg('rfs_tab', '0')
+
+        if args['state'] == 'present':
+            f1 = open("/tmp/dbg1", "w+")
+            qu_num = interactor.determine_qu_num('rx', args['iface'])
+            f1.write(str(qu_num))
+            per_qu_flows = int(args['tab_size']) / int(qu_num)
+            out_para = (qu_num, per_qu_flows)
+
+            changed = interactor.out_cnfg('rfs_tab', args['tab_size'])
+            changed = interactor.out_cnfg('rfs_f_cnt', out_para, iface=args['iface'])
+    else:
+        # rps or xps
+        translator = Translator()
+        associator = Associator(args, module, interactor, translator)
+        changed = associator.distribute()
+
+    # todo: gconsider return qu_masks
+    module.exit_json(changed=changed, **args)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(required=True, choices=['present', 'absent', 'balanced', 'specific']),
+            tech=dict(required=True, choices=['rfs', 'rps', 'xps']),
+            iface=dict(required=True),
+            distribution=dict(required=False, default=None),
+            tab_size=dict(required=False, type='int', default=32768)),
+        supports_check_mode=False
```

Comment:
Hm, no too much, although I deem it somehow superfluous for such a light weight module.